src/components/Board.js

```diff
@@ -5,12 +5,19 @@ import PropTypes from 'prop-types';
 
 
 const generateSquareComponents = (squares, onClickCallback) => {
+  const flatarr = squares.flat();
+  
+  return (flatarr.map(square => {
+    return (
+      <Square id={square.id} value ={square.value} onClickCallback={onClickCallback}></Square>
+    );
+  }));
```

Comment:
You can combine these into a single `flatMap`:

```suggestion
  return (squares.flatMap(square => {
    return (
      <Square id={square.id} value ={square.value} onClickCallback={onClickCallback}></Square>
    );
  }));
```